{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(readr)\n",
    "library(GenomicFeatures)\n",
    "library(DESeq2)\n",
    "library(org.Mm.eg.db)\n",
    "library(rjson)\n",
    "library(tximport)\n",
    "library(DBI)\n",
    "library(rje)\n",
    "library(plyr)\n",
    "library(tidyverse)\n",
    "\n",
    "codedir <- getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  ensembl_stable_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  ensembl_stable_ID = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "######################################## Convert ENSEMBL ID to gene symbols ########################################\n",
    "# Download convert table from: http://useast.ensembl.org/biomart/martview/8c1957c27101a044a318d51140a289e1\n",
    "# Noticed Biomart download error: wrong gene symbol to ensembl ID conversion (e.g. Foxp3 in human reference)\n",
    "\n",
    "cv_mm_file <- '/home/pipkin/references/mm_BioMart_GeneStableID_GeneName.txt'\n",
    "cv_mm_tb <- read_csv(cv_mm_file)\n",
    "\n",
    "cv_hs_file <- '/home/pipkin/references/hs_BioMart_GeneName_GeneStableID.txt'\n",
    "cv_hs_tb <- read_csv(cv_hs_file)\n",
    "\n",
    "matchGN <- function(input, outfilename, cvTb){\n",
    "    colnames(input) <- c(\"ensembl_stable_ID\", colnames(input)[2:length(colnames(input))])\n",
    "    output <- cvTb %>% right_join(input, by=\"ensembl_stable_ID\")\n",
    "    output$ensembl_stable_ID <- NULL\n",
    "    write_csv(output, outfilename)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    }
   ],
   "source": [
    "######################################## Prepare reference ########################################\n",
    "###----- mm reference\n",
    "mm.100.path <- '/home/pipkin/references/GRCm38.100'\n",
    "#setwd(mm.100.path)\n",
    "#txdb <- makeTxDbFromGFF('Mus_musculus.GRCm38.100.gtf')\n",
    "#saveDb(txdb, file='Mus_musculus.GRCm38.100')\n",
    "mmRef <- 'Mus_musculus.GRCm38.100'\n",
    "# Convert transcript ID to gene ID\n",
    "mm_txdb <- loadDb(file.path(mm.100.path, mmRef))\n",
    "mm_k <- keys(mm_txdb, \"GENEID\")\n",
    "mm_res <- AnnotationDbi::select(mm_txdb, mm_k, \"TXNAME\", \"GENEID\")\n",
    "mm_tx2gene <- mm_res[,2:1]\n",
    "\n",
    "###----- hs reference\n",
    "hs.100.path <- '/home/pipkin/references/GRCh38.100'\n",
    "#setwd(hs.100.path)\n",
    "#txdb <- makeTxDbFromGFF('Homo_sapiens.GRCh38.100.gtf')\n",
    "#saveDb(txdb, file='Homo_sapiens.GRCh38.100')\n",
    "hsRef <- 'Homo_sapiens.GRCh38.100'\n",
    "# Convert transcript ID to gene ID\n",
    "hs_txdb <- loadDb(file.path(hs.100.path, hsRef))\n",
    "hs_k <- keys(hs_txdb, \"GENEID\")\n",
    "hs_res <- AnnotationDbi::select(hs_txdb, hs_k, \"TXNAME\", \"GENEID\")\n",
    "hs_tx2gene <- hs_res[,2:1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0. Differential comparison metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Batch = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Celltype1 = \u001b[31mcol_character()\u001b[39m,\n",
      "  Celltype2 = \u001b[31mcol_character()\u001b[39m,\n",
      "  Species = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2016_IMMUNITY_De_Simone\"            \"2017_CELL_REP_Kumar\"               \n",
      "[3] \"2017_NAT_Akondy\"                    \"2019_EJI_Remmerswaal\"              \n",
      "[5] \"2019_NAT_IMMUNOL_Nakayama\"          \"2020_NAT_IMMUNOL_Hayward_GSE118111\"\n",
      "[7] \"2020_NAT_IMMUNOL_Hayward_GSE135504\"\n"
     ]
    }
   ],
   "source": [
    "diff_comp <- '/media/pipkin/Yolanda/SARS2_Sc/z_References/0_Analysis_from_GEO/Differential_Comparisons.csv'\n",
    "diff_comp_df <- read_csv(diff_comp)\n",
    "batches <- unique(diff_comp_df$Batch)\n",
    "\n",
    "print(batches)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Run differential analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "batch_info_dir <- '/media/pipkin/Yolanda/SARS2_Sc/z_References/0_Analysis_from_GEO'\n",
    "salmon_out_dir <- '/media/pipkin/Yolanda/SARS2_Sc/z_References/1_salmon'\n",
    "deseq_out_dir <- '/media/pipkin/Yolanda/SARS2_Sc/z_References/2_DEseq_out'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Description = \u001b[31mcol_character()\u001b[39m,\n",
      "  bio_material = \u001b[31mcol_character()\u001b[39m,\n",
      "  Patient_code = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "21 \n",
      "22 \n",
      "23 \n",
      "24 \n",
      "25 \n",
      "26 \n",
      "27 \n",
      "28 \n",
      "29 \n",
      "30 \n",
      "31 \n",
      "32 \n",
      "33 \n",
      "34 \n",
      "35 \n",
      "36 \n",
      "37 \n",
      "38 \n",
      "39 \n",
      "40 \n",
      "41 \n",
      "42 \n",
      "43 \n",
      "44 \n",
      "45 \n",
      "46 \n",
      "47 \n",
      "48 \n",
      "49 \n",
      "50 \n",
      "51 \n",
      "52 \n",
      "53 \n",
      "54 \n",
      "55 \n",
      "56 \n",
      "57 \n",
      "58 \n",
      "59 \n",
      "60 \n",
      "61 \n",
      "62 \n",
      "63 \n",
      "64 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 8176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "-- replacing outliers and refitting for 1272 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2016_IMMUNITY_De_Simone\"\n",
      "[1] \"Th1\"  \"Th17\"\n",
      "[1] \"Th1\"  \"Treg\"\n",
      "[1] \"Th17\" \"Treg\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  tcell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  tissue = \u001b[31mcol_character()\u001b[39m,\n",
      "  cd69_status = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "21 \n",
      "22 \n",
      "23 \n",
      "24 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 8176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2017_CELL_REP_Kumar\"\n",
      "[1] \"CD4_lung_Pos\" \"CD4_lung_Neg\"\n",
      "[1] \"CD8_lung_Pos\" \"CD8_lung_Neg\"\n",
      "[1] \"CD4_spleen_Pos\" \"CD4_spleen_Neg\"\n",
      "[1] \"CD8_spleen_Pos\" \"CD8_spleen_Neg\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Missing column names filled in: 'X6' [6], 'X7' [7], 'X8' [8], 'X9' [9], 'X10' [10]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Day = \u001b[31mcol_character()\u001b[39m,\n",
      "  Donor = \u001b[31mcol_character()\u001b[39m,\n",
      "  X6 = \u001b[33mcol_logical()\u001b[39m,\n",
      "  X7 = \u001b[33mcol_logical()\u001b[39m,\n",
      "  X8 = \u001b[33mcol_logical()\u001b[39m,\n",
      "  X9 = \u001b[33mcol_logical()\u001b[39m,\n",
      "  X10 = \u001b[33mcol_logical()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 8176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2017_NAT_Akondy\"\n",
      "[1] \"D14\" \"NAV\"\n",
      "[1] \"D14\" \"MEM\"\n",
      "[1] \"MEM\" \"NAV\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  source_name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 8176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2019_EJI_Remmerswaal\"\n",
      "[1] \"DP\" \"EE\"\n",
      "[1] \"DP\" \"MP\"\n",
      "[1] \"DP\" \"TE\"\n",
      "[1] \"EE\" \"MP\"\n",
      "[1] \"EE\" \"TE\"\n",
      "[1] \"MP\" \"TE\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Duplicated column names deduplicated: 'Name' => 'Name_1' [5]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `genotype/variation` = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name_1 = \u001b[31mcol_character()\u001b[39m,\n",
      "  source_name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2019_NAT_IMMUNOL_Nakayama\"\n",
      "[1] \"LungTRreg\" \"LungTRM\"  \n",
      "[1] \"LungTRreg\"  \"Cir_CD44hi\"\n",
      "[1] \"LungTRreg\"  \"Cir_CD44lo\"\n",
      "[1] \"Cir_CD44lo\"         \"Cir_CD44lo_Foxp3KO\"\n",
      "[1] \"LungTRreg\"         \"LungTRreg_Foxp3KO\"\n",
      "[1] \"LungTRM\"         \"LungTRM_Foxp3KO\"\n",
      "[1] \"Cir_CD44hi\" \"Cir_CD44lo\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  organ = \u001b[31mcol_character()\u001b[39m,\n",
      "  source_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  stimulation = \u001b[31mcol_character()\u001b[39m,\n",
      "  tissue = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "-- replacing outliers and refitting for 16 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2020_NAT_IMMUNOL_Hayward_GSE118111\"\n",
      "[1] \"Airway_Trm\" \"Spleen_Tcm\"\n",
      "[1] \"Airway_Trm\"     \"Parenchyma_Trm\"\n",
      "[1] \"Parenchyma_Trm\" \"Spleen_Tcm\"    \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Duplicated column names deduplicated: 'Cell_type' => 'Cell_type_1' [4]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type_1 = \u001b[31mcol_character()\u001b[39m,\n",
      "  organ = \u001b[31mcol_character()\u001b[39m,\n",
      "  stimulation = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 176\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for 2020_NAT_IMMUNOL_Hayward_GSE135504\"\n",
      "[1] \"Airway_Trm\" \"Spleen_Tcm\"\n",
      "[1] \"Lung_intersitium_Trm\" \"Spleen_Tcm\"          \n",
      "[1] \"Lung_vasculature_Trm\" \"Spleen_Tcm\"          \n",
      "[1] \"Airway_Trm\"           \"Lung_intersitium_Trm\"\n",
      "[1] \"Airway_Trm\"           \"Lung_vasculature_Trm\"\n",
      "[1] \"Lung_intersitium_Trm\" \"Lung_vasculature_Trm\"\n"
     ]
    }
   ],
   "source": [
    "for (batch in batches){\n",
    "    batch_info_file <- paste(batch_info_dir, \"/\", batch, \"_SRR_Info.csv\",sep=\"\")\n",
    "    batch_species <- diff_comp_df %>% filter(Batch == batch) %>% .$Species %>% .[1]\n",
    "    batch_info_df <- read_csv(batch_info_file)\n",
    "\n",
    "    diff_comp_df_batch <- diff_comp_df %>% filter(Batch == batch)\n",
    "    use_celltypes <- unique(c(diff_comp_df_batch$Celltype1, diff_comp_df_batch$Celltype2)) \n",
    "\n",
    "    ###----- Sample name & condition setup\n",
    "    colData <- batch_info_df %>% filter(Cell_type %in% use_celltypes) %>% dplyr::select(one_of(c('Run', 'Name', 'Cell_type')))\n",
    "    conditions <- colData$Cell_type\n",
    "    colData$Cell_type <- NULL\n",
    "    colnames(colData) <- c(\"Samples\", 'Cond')\n",
    "    if (! is.subset(use_celltypes, conditions)) { # Check if all conditions are found\n",
    "        print(paste(\"Missing condition in batch\", batch, sep=\" \"))\n",
    "        print(\"--------------------\")\n",
    "        print(\"Conditions to be used: \")\n",
    "        print(use_celltypes)\n",
    "        print(\"--------------------\")\n",
    "        print(\"Conditions found:\")\n",
    "        print(unique(conditions))\n",
    "    }\n",
    "\n",
    "    # Read files\n",
    "    files <- file.path(salmon_out_dir,batch,colData$Samples,\"quant.sf\")\n",
    "    names(files) <- colData$Cond\n",
    "    \n",
    "    # Human or mouse genome\n",
    "    txi <- NULL\n",
    "    cv_tb <- NULL\n",
    "    if (batch_species == \"Human\") {\n",
    "        txi <- tximport(files, type=\"salmon\", tx2gene=hs_tx2gene, ignoreTxVersion = TRUE, dropInfReps = TRUE) # Drop in freps TURE = ignore verison  # Ignore TX verison stringsplits on . \n",
    "        cv_tb <- cv_hs_tb\n",
    "    } else if (batch_species == \"Mouse\") {\n",
    "        txi <- tximport(files, type=\"salmon\", tx2gene=mm_tx2gene, ignoreTxVersion = TRUE, dropInfReps = TRUE) # Drop in freps TURE = ignore verison  # Ignore TX verison stringsplits on . \n",
    "        cv_tb <- cv_mm_tb\n",
    "    } \n",
    "    \n",
    "    # Build sample table\n",
    "    sampleTable <- data.frame(condition = factor(conditions))\n",
    "    rownames(sampleTable) <- colnames(txi$counts)\n",
    "\n",
    "    #import into DESEQ2 framework\n",
    "    dds <- DESeqDataSetFromTximport(txi, sampleTable, ~ condition)\n",
    "    summary(dds)\n",
    "\n",
    "    ###----- Run DEseq\n",
    "    dds <- DESeq(dds)\n",
    "\n",
    "    ###----- Write outputs\n",
    "    print(\"--------------------\")\n",
    "    print(paste(\"Write outputs for \", batch, sep=\"\"))\n",
    "    for (i in c(1:nrow(diff_comp_df_batch))) {\n",
    "        contrast_i <- c(\"condition\", diff_comp_df_batch$Celltype1[i], diff_comp_df_batch$Celltype2[i])\n",
    "        print(contrast_i[2:3])\n",
    "        out_name_i <- paste(batch, \"_\", diff_comp_df_batch$Name[i], \".csv\", sep=\"\")\n",
    "        out_name_i <- file.path(deseq_out_dir, out_name_i)\n",
    "        out_name_i_gn <- gsub(\".csv\",\"_gn.csv\",  out_name_i)\n",
    "        results <- as_tibble(results(dds, contrast = contrast_i), rownames='ensembl_id')\n",
    "        write_csv(results, out_name_i)\n",
    "        matchGN(results, out_name_i_gn, cv_tb)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
