{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(readr)\n",
    "library(GenomicFeatures)\n",
    "library(DESeq2)\n",
    "library(org.Mm.eg.db)\n",
    "library(rjson)\n",
    "library(tximport)\n",
    "library(DBI)\n",
    "library(rje)\n",
    "library(plyr)\n",
    "\n",
    "codedir <- getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  ensembl_stable_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "######################################## Convert ENSEMBL ID to gene symbols ########################################\n",
    "# Download convert table from:http://useast.ensembl.org/biomart/martview/8c1957c27101a044a318d51140a289e1\n",
    "\n",
    "cv_file <- '/home/pipkin/references/mm_BioMart_GeneStableID_GeneName.txt'\n",
    "cv_tb <- read_csv(cv_file)\n",
    "\n",
    "matchGN <- function(input, outfilename, cvTb=cv_tb){\n",
    "    colnames(input) <- c(\"ensembl_stable_ID\", colnames(input)[2:length(colnames(input))])\n",
    "    output <- cvTb %>% right_join(input, by=\"ensembl_stable_ID\")\n",
    "    output$ensembl_stable_ID <- NULL\n",
    "    write_csv(output, outfilename)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0. Prepare reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    }
   ],
   "source": [
    "###--- Make reference\n",
    "#txdb <- makeTxDbFromGFF('/home/pipkin/references/GRCm38.99/Mus_musculus.GRCm38.99.gtf')\n",
    "#saveDb(txdb, file='/home/pipkin/references/GRCm38.99/Mus_musculus.GRCm38.99')\n",
    "mmRef <- '/home/pipkin/references/GRCm38.99/Mus_musculus.GRCm38.99'\n",
    "\n",
    "###--- Convert transcript ID to gene ID\n",
    "txdb <- loadDb(mmRef)\n",
    "k <- keys(txdb, \"GENEID\")\n",
    "res <- AnnotationDbi::select(txdb, k, \"TXNAME\", \"GENEID\")\n",
    "tx2gene <- res[,2:1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Differential comparison metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Batch = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Celltype1 = \u001b[31mcol_character()\u001b[39m,\n",
      "  Celltype2 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"GSE72408\"  \"GSE68056\"  \"GSE132110\" \"GSE70813\" \n"
     ]
    }
   ],
   "source": [
    "diff_comp_df <- read_csv(\"Differential_Comparisons.csv\")\n",
    "batches <- unique(diff_comp_df$Batch)\n",
    "\n",
    "print(batches)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Run differential analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "salmon_out_dir <- '/media/pipkin/ROCKET-PRO/GEO_RNAseq/1_salmon_out'\n",
    "deseq_out_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DEseq_out'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Info = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "21 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 175\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for GSE72408\"\n",
      "[1] \"WT\"     \"Zeb2KO\"\n",
      "[1] \"WT\"      \"Tbx21KO\"\n",
      "[1] \"Klrg1hi_WT\"     \"Klrg1hi_Zeb2KO\"\n",
      "[1] \"Klrg1hi_WT\"      \"Klrg1hi_Tbx21OE\"\n",
      "[1] \"Klrg1hi_Zeb2KO\"         \"Klrg1hi_Zeb2KO_Tbx21OE\"\n",
      "[1] \"Klrg1hi_Tbx21OE\"        \"Klrg1hi_Zeb2KO_Tbx21OE\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Description = \u001b[31mcol_character()\u001b[39m,\n",
      "  Info = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 175\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "-- replacing outliers and refitting for 566 genes\n",
      "-- DESeq argument 'minReplicatesForReplace' = 7 \n",
      "-- original counts are preserved in counts(dds)\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for GSE68056\"\n",
      "[1] \"WT\"      \"Tbx21KO\"\n",
      "[1] \"WT\"       \"Blimp1KO\"\n",
      "[1] \"WT\"      \"Il2raKO\"\n",
      "[1] \"Il2raKO\"          \"Il2raKO_Blimp1KO\"\n",
      "[1] \"Blimp1KO\"         \"Il2raKO_Blimp1KO\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Info = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 175\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for GSE132110\"\n",
      "[1] \"Tim3pos\" \"Tim3neg\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Run = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_description = \u001b[31mcol_character()\u001b[39m,\n",
      "  Tissue = \u001b[31mcol_character()\u001b[39m,\n",
      "  Genotype = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cell_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Name = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "21 \n",
      "22 \n",
      "23 \n",
      "24 \n",
      "25 \n",
      "26 \n",
      "27 \n",
      "28 \n",
      "\n",
      "\n",
      "transcripts missing from tx2gene: 175\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not an warning or error]\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"--------------------\"\n",
      "[1] \"Write outputs for GSE70813\"\n",
      "[1] \"WT\"       \"Hobit_KO\"\n",
      "[1] \"WT\"        \"Blimp1_KO\"\n",
      "[1] \"Hobit_KO\"        \"Hobit-Blimp1_KO\"\n",
      "[1] \"Blimp1_KO\"       \"Hobit-Blimp1_KO\"\n",
      "[1] \"Sp_TCM\" \"Sp_TEM\"\n",
      "[1] \"Liver_TEM\" \"Liver_TRM\"\n",
      "[1] \"Sp_TCM\"    \"Liver_TRM\"\n",
      "[1] \"Sp_TCM\"    \"Liver_TEM\"\n",
      "[1] \"Sp_TCM\"   \"Skin_TRM\"\n",
      "[1] \"Sp_TCM\"  \"Gut_TRM\"\n",
      "[1] \"Sp_TEM\"    \"Liver_TRM\"\n",
      "[1] \"Sp_TEM\"    \"Liver_TEM\"\n",
      "[1] \"Sp_TEM\"   \"Skin_TRM\"\n",
      "[1] \"Sp_TEM\"  \"Gut_TRM\"\n"
     ]
    }
   ],
   "source": [
    "for (batch in batches){\n",
    "    batch_info_file <- paste(codedir, \"/\", batch, \"_SRR_Info.csv\",sep=\"\")\n",
    "    batch_info_df <- read_csv(batch_info_file)\n",
    "\n",
    "    diff_comp_df_batch <- diff_comp_df %>% filter(Batch == batch)\n",
    "    use_celltypes <- unique(c(diff_comp_df_batch$Celltype1, diff_comp_df_batch$Celltype2)) \n",
    "\n",
    "    ###----- Sample name & condition setup\n",
    "    colData <- batch_info_df %>% filter(Cell_type %in% use_celltypes) %>% dplyr::select(one_of(c('Run', 'Name', 'Cell_type')))\n",
    "    conditions <- colData$Cell_type\n",
    "    colData$Cell_type <- NULL\n",
    "    colnames(colData) <- c(\"Samples\", 'Cond')\n",
    "    if (! is.subset(use_celltypes, conditions)) { # Check if all conditions are found\n",
    "        print(paste(\"Missing condition in batch\", batch, sep=\" \"))\n",
    "        print(\"--------------------\")\n",
    "        print(\"Conditions to be used: \")\n",
    "        print(use_celltypes)\n",
    "        print(\"--------------------\")\n",
    "        print(\"Conditions found:\")\n",
    "        print(unique(conditions))\n",
    "    }\n",
    "\n",
    "    # Read files\n",
    "    files <- file.path(salmon_out_dir,colData$Samples,\"quant.sf\")\n",
    "    names(files) <- colData$Cond\n",
    "    txi <- tximport(files, type=\"salmon\", tx2gene=tx2gene, ignoreTxVersion = TRUE, dropInfReps = TRUE) # Drop in freps TURE = ignore verison  # Ignore TX verison stringsplits on . \n",
    "\n",
    "    # Build sample table\n",
    "    sampleTable <- data.frame(condition = factor(conditions))\n",
    "    rownames(sampleTable) <- colnames(txi$counts)\n",
    "\n",
    "    #import into DESEQ2 framework\n",
    "    dds <- DESeqDataSetFromTximport(txi, sampleTable, ~ condition)\n",
    "    summary(dds)\n",
    "\n",
    "    ###----- Run DEseq\n",
    "    dds <- DESeq(dds)\n",
    "\n",
    "    ###----- Write outputs\n",
    "    print(\"--------------------\")\n",
    "    print(paste(\"Write outputs for \", batch, sep=\"\"))\n",
    "    for (i in c(1:nrow(diff_comp_df_batch))) {\n",
    "        contrast_i <- c(\"condition\", diff_comp_df_batch$Celltype1[i], diff_comp_df_batch$Celltype2[i])\n",
    "        print(contrast_i[2:3])\n",
    "        out_name_i <- paste(batch, \"_\", diff_comp_df_batch$Name[i], \".csv\", sep=\"\")\n",
    "        out_name_i <- file.path(deseq_out_dir, out_name_i)\n",
    "        out_name_i_gn <- gsub(\".csv\",\"_gn.csv\",  out_name_i)\n",
    "        results <- as_tibble(results(dds, contrast = contrast_i), rownames='ensembl_id')\n",
    "        write_csv(results, out_name_i)\n",
    "        matchGN(results, out_name_i_gn)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Create GSEA signature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "wk_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/GSEA'\n",
    "setwd(wk_dir)\n",
    "\n",
    "deseq_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DEseq_out'\n",
    "gn_DEseq_files <- list.files(deseq_dir,pattern=\"*gn.csv\", full.name=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "gs_name <- c()\n",
    "gene_symbol <- c()\n",
    "\n",
    "file_i <- gn_DEseq_files[1]\n",
    "\n",
    "for (file_i in gn_DEseq_files){\n",
    "    file_i_name_simp <- basename(file_i)\n",
    "    file_i_name_simp <- gsub(\"_gn.csv\", \"\", file_i_name_simp)\n",
    "    file_i_name_simp <- gsub(\"_vs_\", \"_\", file_i_name_simp)\n",
    "    file_i_name_simp <- gsub(\"---\", \"_\", file_i_name_simp)\n",
    "    file_i_name_simp_vec <- unlist(strsplit(file_i_name_simp, \"_\"))\n",
    "    file_i_name_cp <- tail(file_i_name_simp_vec, 2)\n",
    "\n",
    "    gs_name_1 <- paste(gsub(\"_gn.csv\", \"\", basename(file_i)), \"---\", file_i_name_cp[1], sep=\"\")\n",
    "    gs_name_2 <- paste(gsub(\"_gn.csv\", \"\", basename(file_i)), \"---\", file_i_name_cp[2], sep=\"\")\n",
    "\n",
    "    df_i <- read_csv(file_i)\n",
    "    \n",
    "    df_i_up <- df_i %>% filter(padj <= 0.05) %>% filter(log2FoldChange >= 1)\n",
    "    df_i_dn <- df_i %>% filter(padj <= 0.05) %>% filter(log2FoldChange <= -1)\n",
    "\n",
    "    gene_symbol <- c(gene_symbol, df_i_up$gene_name)\n",
    "    gs_name <- c(gs_name, rep(gs_name_1, nrow(df_i_up)))\n",
    "\n",
    "    gene_symbol <- c(gene_symbol, df_i_dn$gene_name)\n",
    "    gs_name <- c(gs_name, rep(gs_name_2, nrow(df_i_dn)))\n",
    "}\n",
    "\n",
    "### Create dataframe and summarize\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"GEO_signatures.csv\")\n",
    "write_csv(count_df, \"GEO_sigantures_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Select signatures to use*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_number = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "use_sig <- 'GEO_sigantures_use.csv'\n",
    "use_sig_df <- read_csv(use_sig)\n",
    "\n",
    "gsea_df <- read_csv(\"GEO_signatures.csv\")\n",
    "gsea_df_use <- gsea_df %>% filter(gs_name %in% use_sig_df$gs_name)\n",
    "\n",
    "write_csv(gsea_df_use, \"GEO_signatures_use.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Create signatures for DE from supplementary data of publications"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "wk_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/GSEA'\n",
    "setwd(wk_dir)\n",
    "\n",
    "deseq_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WangDapeng - Runx3\n",
    "- Use day 5 genes and split EE and TE signatures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"Ccr2\"          \"Gzma\"          \"Clec12a\"       \"Mt1\"          \n",
      " [5] \"P2ry12\"        \"Cd244\"         \"Atp8b4\"        \"Tcrg-C1\"      \n",
      " [9] \"H2-Q10\"        \"Chst11\"        \"Gcnt1\"         \"1500009L16Rik\"\n",
      "[13] \"Gm13139\"       \"Mt2\"           \"Tmem171\"       \"Anxa1\"        \n",
      "[17] \"Gm15518\"       \"Klrk1\"         \"Wdr95\"         \"Galnt3\"       \n",
      "[21] \"Adam8\"         \"AA467197\"      \"Epas1\"         \"Tmem41a\"      \n",
      "[25] \"Tle1\"          \"Lilr4b\"        \"Il15\"          \"Slc7a3\"       \n",
      "[29] \"Gzmb\"          \"Amica1\"        \"Aunip\"         \"Trim36\"       \n",
      "[33] \"Ccbl2\"         \"Prss30\"        \"Asns\"          \"Cdh17\"        \n",
      "[37] \"Ptprk\"         \"Maoa\"          \"Pls1\"          \"Bco1\"         \n",
      "[41] \"Tnfsf9\"        \"Gpx8\"          \"Sema4f\"        \"Tspan4\"       \n",
      "[45] \"Il2ra\"         \"Dgkh\"          \"Dcbld2\"        \"Srgap3\"       \n",
      "[49] \"Fasl\"          \"Filip1\"        \"Cd96\"          \"Celf4\"        \n",
      "[53] \"Trbv14\"        \"2010204K13Rik\" \"Spats2\"        \"Ifitm2\"       \n",
      "[57] \"Tnfrsf9\"       \"Adamts14\"      \"4930509E16Rik\" \"Med12l\"       \n",
      "[61] \"Tnni3\"         \"Msc\"           \"Rsg1\"          \"1700026L06Rik\"\n",
      "[65] \"Tcrg-V4\"       \"Batf3\"         \"Ifitm1\"        \"Xcl1\"         \n",
      "[69] \"Qpct\"          \"Gpr34\"         \"Cxcr6\"         \"Tcrg-C2\"      \n",
      "[73] \"Gm6169\"        \"Plpp1\"         \"Il31ra\"        \"Ninj2\"        \n",
      "[77] \"Fcer1g\"        \"Prune2\"        \"Spry2\"         \"Acer1\"        \n",
      "[81] \"Tcrg-C4\"       \"Bsn\"           \"Lat2\"          \"Lilrb4a\"      \n",
      "[85] \"Tubb3\"         \"Slc6a13\"      \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  [1] \"Ccr2\"          \"Gm13157\"       \"Il18r1\"        \"Slc30a4\"      \n",
      "  [5] \"Swap70\"        \"Nebl\"          \"Vps50\"         \"Gcnt1\"        \n",
      "  [9] \"Gm13139\"       \"Klf8\"          \"Armcx4\"        \"Hook1\"        \n",
      " [13] \"Marveld2\"      \"P2ry12\"        \"Bspry\"         \"Gm19261\"      \n",
      " [17] \"Tle1\"          \"Cd55\"          \"Reck\"          \"Eif2ak3\"      \n",
      " [21] \"Galnt3\"        \"Epas1\"         \"Cd244\"         \"Cd24a\"        \n",
      " [25] \"Clec12a\"       \"Fam84b\"        \"Tob1\"          \"Slc52a3\"      \n",
      " [29] \"Ptpn3\"         \"1500009L16Rik\" \"Mpp7\"          \"Amica1\"       \n",
      " [33] \"Lamc2\"         \"Spats2\"        \"Celf4\"         \"Ogfrl1\"       \n",
      " [37] \"Prss16\"        \"Sfmbt2\"        \"Nt5e\"          \"Tmem221\"      \n",
      " [41] \"9230110C19Rik\" \"St14\"          \"Pgap1\"         \"Mt2\"          \n",
      " [45] \"Dhrs13\"        \"Klf4\"          \"Pak6\"          \"Mtss1\"        \n",
      " [49] \"Adam8\"         \"1110032F04Rik\" \"Naf1\"          \"Prss30\"       \n",
      " [53] \"Cyp2s1\"        \"Gm37169\"       \"Usp53\"         \"Spin2c\"       \n",
      " [57] \"Dnajc6\"        \"Trim36\"        \"Ggt1\"          \"Tex15\"        \n",
      " [61] \"Gtf2ird1\"      \"Gnaq\"          \"Epcam\"         \"Maats1\"       \n",
      " [65] \"Prune2\"        \"Tfcp2l1\"       \"Eya2\"          \"AA474331\"     \n",
      " [69] \"Dsp\"           \"Irak1bp1\"      \"Fbxo27\"        \"Fut7\"         \n",
      " [73] \"Spry2\"         \"Sowahc\"        \"Acad11\"        \"Gm37289\"      \n",
      " [77] \"Mmp10\"         \"Filip1\"        \"Sept10\"        \"Gm24453\"      \n",
      " [81] \"Prrg1\"         \"Fam20a\"        \"Gm15832\"       \"Atp8a2\"       \n",
      " [85] \"Mir3473a\"      \"Dyrk3\"         \"Cep170b\"       \"Rnf128\"       \n",
      " [89] \"Man1c1\"        \"Ift88\"         \"Zfp383\"        \"Trbv3\"        \n",
      " [93] \"Gm37271\"       \"Gm26707\"       \"Gm28942\"       \"Gm5611\"       \n",
      " [97] \"Pard6b\"        \"Crb3\"          \"Mmp16\"         \"Bicd1\"        \n",
      "[101] \"Ttn\"           \"Gm13075\"       \"Tubb3\"         \"Tha1\"         \n",
      "[105] \"Gzmc\"          \"Il2ra\"         \"Ldhb\"          \"Gm26614\"      \n",
      "[109] \"Snhg14\"        \"Muc1\"          \"Rnf17\"         \"Cxadr\"        \n",
      "[113] \"Scd1\"          \"Styk1\"         \"Rny1\"          \"Tns1\"         \n",
      "[117] \"P2ry13\"        \"Lrp11\"         \"Ifitm1\"        \"Rsph10b\"      \n",
      "[121] \"Cdkl2\"         \"Gm37881\"       \"S100a9\"        \"Hspg2\"        \n",
      "[125] \"Bbof1\"         \"Fbxo44\"        \"March8\"        \"Il1r2\"        \n",
      "[129] \"Dmwd\"          \"Gm29170\"       \"S100a5\"        \"Zfp608\"       \n",
      "[133] \"Acot6\"        \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"Ccr2\"          \"Hsph1\"         \"Gm13157\"       \"Gm13139\"      \n",
      " [5] \"Gm4208\"        \"Swap70\"        \"Tktl1\"         \"Slc15a2\"      \n",
      " [9] \"Padi2\"         \"Gm26614\"       \"Galnt3\"        \"Col23a1\"      \n",
      "[13] \"Itga1\"         \"AA467197\"      \"Xcr1\"          \"Amica1\"       \n",
      "[17] \"Ier3\"          \"Itgam\"         \"Ccdc180\"       \"Dgkh\"         \n",
      "[21] \"Paqr7\"         \"Trdv4\"         \"Gm28942\"       \"Kcnq1\"        \n",
      "[25] \"Maoa\"          \"Kcnk6\"         \"Med12l\"        \"Gm42722\"      \n",
      "[29] \"Jrk\"           \"Spef2\"         \"Mal\"           \"Plod1\"        \n",
      "[33] \"P2ry13\"        \"Rapgef3\"       \"Col7a1\"        \"Adam11\"       \n",
      "[37] \"Celsr3\"        \"Ret\"           \"Rab6b\"         \"Mybpc3\"       \n",
      "[41] \"Neurl1b\"       \"Slc4a5\"        \"Col2a1\"        \"8030474K03Rik\"\n",
      "[45] \"Bahcc1\"        \"Gm26811\"       \"Kif17\"         \"BC024139\"     \n",
      "[49] \"Mroh4\"         \"Aire\"          \"Nova1\"         \"Qpct\"         \n",
      "[53] \"Cep170b\"       \"Nlrp6\"         \"Spire1\"        \"Clcn2\"        \n",
      "[57] \"Lzts3\"         \"Nfasc\"         \"Aatk\"          \"Robo3\"        \n",
      "[61] \"Spns2\"         \"Gpihbp1\"      \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  [1] \"Ccr2\"          \"Ptpn4\"         \"Cx3cr1\"        \"Id2\"          \n",
      "  [5] \"Tktl1\"         \"Anxa1\"         \"Wdr95\"         \"Gzma\"         \n",
      "  [9] \"Gzmb\"          \"Itgax\"         \"Gm13157\"       \"Gm4208\"       \n",
      " [13] \"Itga1\"         \"Fasl\"          \"Sytl2\"         \"AA467197\"     \n",
      " [17] \"Glrx\"          \"Rasl11a\"       \"P2ry12\"        \"Klrk1\"        \n",
      " [21] \"Cd86\"          \"Gm15518\"       \"2610020H08Rik\" \"Tmem163\"      \n",
      " [25] \"Prf1\"          \"F2rl2\"         \"Klrg1\"         \"G0s2\"         \n",
      " [29] \"Entpd1\"        \"Cib2\"          \"5430427O19Rik\" \"Ccl3\"         \n",
      " [33] \"Gm10522\"       \"E230016K23Rik\" \"Gm26614\"       \"Ccdc180\"      \n",
      " [37] \"Gm15473\"       \"Txndc5\"        \"Smpdl3b\"       \"Prdm1\"        \n",
      " [41] \"Rgcc\"          \"Trdv4\"         \"Gm13139\"       \"Lilr4b\"       \n",
      " [45] \"Ccl4\"          \"Chil5\"         \"Lpar6\"         \"Crybb1\"       \n",
      " [49] \"2310031A07Rik\" \"Socs2\"         \"Galnt3\"        \"Klrd1\"        \n",
      " [53] \"Cxcr6\"         \"Ier3\"          \"Lamc1\"         \"Xcr1\"         \n",
      " [57] \"Serpini1\"      \"Ccrl2\"         \"Fcgr2b\"        \"Gm12253\"      \n",
      " [61] \"Ccl5\"          \"Gm11707\"       \"Akr1c13\"       \"Cd48\"         \n",
      " [65] \"Havcr2\"        \"Itgam\"         \"Lmntd1\"        \"Actn2\"        \n",
      " [69] \"2900041M22Rik\" \"Gstm5\"         \"Il15\"          \"Sgk3\"         \n",
      " [73] \"1500009L16Rik\" \"Dkkl1\"         \"Lxn\"           \"Lgals3\"       \n",
      " [77] \"Ccl9\"          \"Nhsl2\"         \"Serp2\"         \"Amica1\"       \n",
      " [81] \"Trbv3\"         \"Pls1\"          \"Qpct\"          \"Rsg1\"         \n",
      " [85] \"Cldnd1\"        \"Dgkh\"          \"Klrc1\"         \"Slc15a2\"      \n",
      " [89] \"Gm6712\"        \"Tcrg-C1\"       \"Gzmc\"          \"Lilrb4a\"      \n",
      " [93] \"Ica1\"          \"Tcrg-V4\"       \"Trbv1\"         \"Dhrs9\"        \n",
      " [97] \"Gm42921\"       \"Tnfsf9\"        \"Adamts14\"      \"Ly6g5b\"       \n",
      "[101] \"Cdh17\"         \"Selm\"          \"Gm42722\"       \"P2ry14\"       \n",
      "[105] \"Gm28942\"       \"Mal\"           \"Ppm1n\"         \"Pabpc1l\"      \n",
      "[109] \"Maoa\"          \"Trbv14\"        \"Edaradd\"       \"Cacnb4\"       \n",
      "[113] \"Arl11\"         \"Gm10640\"       \"Fam129b\"       \"Trav7-5\"      \n",
      "[117] \"Gm11714\"       \"Cfap54\"        \"Tmprss13\"      \"Acoxl\"        \n",
      "[121] \"Nkapl\"         \"2700012I20Rik\" \"5830416I19Rik\" \"Ribc1\"        \n",
      "[125] \"Efcab6\"        \"Trdv5\"         \"A230056P14Rik\" \"Dpep2\"        \n",
      "[129] \"Tmem37\"        \"Slc35e3\"       \"Fam3b\"        \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  baseMean = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  lfcSE = \u001b[32mcol_double()\u001b[39m,\n",
      "  stat = \u001b[32mcol_double()\u001b[39m,\n",
      "  pvalue = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"Gm13157\" \"Il7r\"    \"Tcrg-C1\" \"Ccr2\"    \"Amica1\"  \"Cd55\"    \"Gm4208\" \n",
      " [8] \"Wfikkn2\" \"Tktl1\"   \"Gm13139\" \"Aunip\"   \"Ier3\"    \"Rnf144a\" \"Gm42722\"\n",
      "[15] \"Slc15a2\" \"Nt5e\"    \"Trbv1\"   \"Tmem221\" \"Prss16\"  \"Msc\"     \"Wbscr27\"\n",
      "[22] \"Paqr7\"   \"Tmem171\" \"Pabpc1l\" \"Gstm5\"   \"Qpct\"   \n"
     ]
    }
   ],
   "source": [
    "DW_gn_DEseq_files <- list.files(deseq_dir,pattern=\"WangDapeng*\", full.name=TRUE)\n",
    "\n",
    "Runx3_promoted <- c()\n",
    "Runx3_repressed <- c()\n",
    "\n",
    "for (file_i in DW_gn_DEseq_files){\n",
    "    i_df <- read_csv(file_i)\n",
    "    i_df_runx3_promoted <- i_df %>% dplyr::filter(padj <= 0.05) %>% \n",
    "    dplyr::filter(log2FoldChange <= -1.5) %>% .$gene_name\n",
    "    i_df_runx3_repressed <- i_df %>% dplyr::filter(padj <= 0.05) %>% \n",
    "    dplyr::filter(log2FoldChange >= 1.5) %>% .$gene_name\n",
    "    print(i_df_runx3_promoted)\n",
    "    \n",
    "    Runx3_promoted <- c(Runx3_promoted, i_df_runx3_promoted)\n",
    "    Runx3_repressed <- c(Runx3_promoted, i_df_runx3_repressed)\n",
    "}\n",
    "\n",
    "gs_name <- c(rep(\"Runx3_promoted\", length(Runx3_promoted)), rep(\"Runx3_repressed\", length(Runx3_repressed)))\n",
    "gene_symbol <- c(Runx3_promoted, Runx3_repressed)\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"WangDapeng_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"WangDapeng_GSEA_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### BeltraJeanChristophe - Tex subsets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Texint vs Texprog1` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texint vs Texprog2` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texint vs Texterm` = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Texprog1 vs Texprog2` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texprog1 vs Texint` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texprog1 vs Texterm` = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Texprog2 vs Texprog1` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texprog2 vs Texint` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texprog2 vs Texterm` = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Texterm vs Texprog1` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texterm vs Texprog2` = \u001b[32mcol_double()\u001b[39m,\n",
      "  `Texterm vs Texint` = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "BeltraJeanChristophe_subset_files <- list.files(deseq_dir,pattern=\"BeltraJeanChristophe*\", full.name=TRUE)\n",
    "\n",
    "gs_name <- c()\n",
    "gene_symbol <- c()\n",
    "\n",
    "for (file_i in BeltraJeanChristophe_subset_files){\n",
    "    i_df <- read_csv(file_i) %>% mutate(rowSum = rowSums(.[2:4]))\n",
    "    \n",
    "    i_name <- gsub(\".csv\",\"\",basename(file_i))\n",
    "    i_name <- tail(unlist(strsplit(i_name, \"_\")), 1)\n",
    "    i_name_up <- paste(i_name, \"up\", sep=\"_\")\n",
    "    i_name_dn <- paste(i_name, \"dn\", sep=\"_\")\n",
    "    \n",
    "    i_up <- i_df %>% filter(rowSum >= 2) %>% .$gene_name\n",
    "    i_dn <- i_df %>% filter(rowSum <= -2) %>% .$gene_name\n",
    "    \n",
    "    gs_name <- c(gs_name, rep(i_name_up, length(i_up)), rep(i_name_dn, length(i_dn)))\n",
    "    gene_symbol <- c(gene_symbol, i_up, i_dn)\n",
    "}\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"BeltraJeanChristophe_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"BeltraJeanChristophe_GSEA_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KhanOmar - Tox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m,\n",
      "  padj = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "KhanOmar_files <- list.files(deseq_dir,pattern=\"KhanOmar*\", full.name=TRUE)\n",
    "\n",
    "Tox_KO_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/KhanOmar_ToxKO_vs_WT_gn.csv'\n",
    "Tox_OE_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/KhanOmar_ToxOE_vs_WT_gn.csv'\n",
    "\n",
    "gs_name <- c()\n",
    "gene_symbol <- c()\n",
    "\n",
    "Tox_KO_up <- read_csv(Tox_KO_file) %>% dplyr::filter(padj <= 0.05) %>% dplyr::filter(log2FoldChange >= 1) %>% .$gene_name\n",
    "Tox_KO_dn <- read_csv(Tox_KO_file) %>% dplyr::filter(padj <= 0.05) %>% dplyr::filter(log2FoldChange <= -1) %>% .$gene_name\n",
    "Tox_OE_up <- read_csv(Tox_OE_file) %>% dplyr::filter(padj <= 0.05) %>% dplyr::filter(log2FoldChange >= 1) %>% .$gene_name\n",
    "Tox_OE_dn <- read_csv(Tox_OE_file) %>% dplyr::filter(padj <= 0.05) %>% dplyr::filter(log2FoldChange <= -1) %>% .$gene_name\n",
    "\n",
    "Tox_promoted <- c(Tox_KO_dn, Tox_OE_up)\n",
    "Tox_repressed <- c(Tox_KO_up, Tox_OE_dn)\n",
    "\n",
    "gs_name <- c(rep(\"Tox_promoted\", length(Tox_promoted)), rep(\"Tox_repressed\", length(Tox_repressed)))\n",
    "gene_symbol <- c(Tox_promoted, Tox_repressed)\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"KhanOmar_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"KhanOmar_GSEA_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WuTuoqi - Tcf7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `p-value` = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `p-value` = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  `p-value` = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "`as.tibble()` is deprecated as of tibble 2.0.0.\n",
      "Please use `as_tibble()` instead.\n",
      "The signature and semantics have changed, see `?as_tibble`.\n",
      "\u001b[90mThis warning is displayed once every 8 hours.\u001b[39m\n",
      "\u001b[90mCall `lifecycle::last_warnings()` to see where this warning was generated.\u001b[39m\n"
     ]
    }
   ],
   "source": [
    "Wu_DEseq_files <- list.files(deseq_dir,pattern=\"WuTuoqi*\", full.name=TRUE)\n",
    "\n",
    "Tcf1oe_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/WuTuoqi_TCF1OE_vs_WT_gn.csv'\n",
    "Tcf7ko_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/WuTuoqi_Tcf7KO_vs_WT_gn.csv'\n",
    "Tim3Blimp1_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/WuTuoqi_Tim3negBlimp1neg_vs_Tim3posBlimp1pos_gn.csv'\n",
    "\n",
    "Tcf1oe_tb <- read_csv(Tcf1oe_file)\n",
    "Tcf7ko_tb <- read_csv(Tcf7ko_file)\n",
    "Tim3Blimp1_tb <- read_csv(Tim3Blimp1_file)\n",
    "\n",
    "Tcf1_promoted <- Tcf1oe_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange >= 1) %>% .$gene_name\n",
    "Tcf1_repressed <- Tcf1oe_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange <= -1) %>% .$gene_name\n",
    "Tcf7ko_promoted <- Tcf7ko_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange >= 1) %>% .$gene_name\n",
    "Tcf7ko_repressed <- Tcf7ko_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange <= -1) %>% .$gene_name\n",
    "\n",
    "Tcf1_promoted <- c(Tcf1_promoted, Tcf7ko_repressed)\n",
    "Tcf1_repressed <- c(Tcf1_repressed, Tcf7ko_promoted)\n",
    "\n",
    "Tim3negBlimp1neg <- Tim3Blimp1_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange >= 1) %>% .$gene_name\n",
    "Tim3posBlimp1pos <- Tim3Blimp1_tb %>% dplyr::filter(`p-value` <= 0.05) %>% dplyr::filter(log2FoldChange <= -1) %>% .$gene_name\n",
    "\n",
    "gs_name <- c(rep(\"Tcf7_promoted\", length(Tcf1_promoted)), \n",
    "            rep(\"Tcf7_repressed\", length(Tcf1_repressed)),\n",
    "            rep(\"Tim3negBlimp1neg\", length(Tim3negBlimp1neg)),\n",
    "            rep(\"Tim3posBlimp1pos\", length(Tim3posBlimp1pos)))\n",
    "\n",
    "gene_symbol <- c(Tcf1_promoted, Tcf1_repressed, Tim3negBlimp1neg, Tim3posBlimp1pos)\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"WuTuoqi_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"WuTuoqi_GSEA_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### KurdNadia - TRM v.s. TE precursor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gene_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  Cluster_16_Mean_TPM = \u001b[32mcol_double()\u001b[39m,\n",
      "  Cluster_20_Mean_TPM = \u001b[32mcol_double()\u001b[39m,\n",
      "  log10pval = \u001b[32mcol_double()\u001b[39m,\n",
      "  log2FoldChange = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "1 parsing failure.\n",
      " row       col expected actual                                                                                                             file\n",
      "3325 log10pval a double    N/A '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/KurdNadia_Cluster16_vs_Cluster20_gn.csv'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "KurdNadia_file <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/DE_from_SUP/KurdNadia_Cluster16_vs_Cluster20_gn.csv'\n",
    "KN_df <- read_csv(KurdNadia_file)\n",
    "\n",
    "TRMbiased <- KN_df %>% dplyr::filter(log10pval < -1.3) %>% dplyr::filter(log2FoldChange >= 0.5) %>% .$gene_name\n",
    "TEbiased <- KN_df %>% dplyr::filter(log10pval < -1.3) %>% dplyr::filter(log2FoldChange <= -0.5) %>% .$gene_name\n",
    "\n",
    "gs_name <- c(rep(\"TRM_biased\", length(TRMbiased)), rep(\"TE_biased\", length(TEbiased)))\n",
    "gene_symbol <- c(TRMbiased, TEbiased)\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"KurdNadia_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"KurdNadia_GSEA_count.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Merge all signatures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  gs_name = \u001b[31mcol_character()\u001b[39m,\n",
      "  gene_symbol = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wk_dir <- '/media/pipkin/Yolanda/Exp391_Acute-Chronic_SC/z_References/GSEA'\n",
    "setwd(wk_dir)\n",
    "\n",
    "signature_files <- list.files(wk_dir,pattern=\"*.csv\", full.name=TRUE)\n",
    "\n",
    "gs_name <- c()\n",
    "gene_symbol <- c()\n",
    "for (file_i in signature_files){\n",
    "    i_df <- read_csv(file_i)\n",
    "    gs_name <- c(gs_name, i_df$gs_name)\n",
    "    gene_symbol <- c(gene_symbol, i_df$gene_symbol)\n",
    "}\n",
    "\n",
    "gsea_df <- as.tibble(data.frame(gs_name, gene_symbol))\n",
    "count_df <- count(gsea_df$gs_name)\n",
    "colnames(count_df) <- c(\"gs_name\", \"gene_number\")\n",
    "\n",
    "write_csv(gsea_df, \"Combined_GEO_GSEA_signatures.csv\")\n",
    "write_csv(count_df, \"Combined_GEO_GSEA_count.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
